x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportDeleteTrainedModelAction.java

```diff
@@ -228,6 +244,39 @@ private void deleteModel(DeleteTrainedModelAction.Request request, ClusterState
         }
     }
 
+    protected boolean modelExists(String modelId) {
+        CountDownLatch latch = new CountDownLatch(1);
+        final AtomicBoolean modelExists = new AtomicBoolean(false);
+
+        ActionListener<TrainedModelConfig> trainedModelListener = new ActionListener<>() {
+            @Override
+            public void onResponse(TrainedModelConfig config) {
+                modelExists.set(true);
+                latch.countDown();
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                logger.error("Failed to retrieve model {}: {}", modelId, e.getMessage(), e);
+                latch.countDown();
+            }
+        };
+
+        trainedModelProvider.getTrainedModel(modelId, GetTrainedModelsAction.Includes.empty(), null, trainedModelListener);
+
+        try {
+            boolean latchReached = latch.await(5, TimeUnit.SECONDS);
```

Comment:
I don't think we want a 5 second timeout here which will throw an exception. If this timeout occurs, I don't think it will be any clearer what happened for the end user.